src/smt/optimization_solver.h

```diff
@@ -80,38 +90,350 @@ class Objective
   /** The node associated to the term that was used to construct the objective.
    * **/
   Node d_node;
-  };
-
-  /**
-   * A solver for optimization queries.
-   * 
-   * This class is responsible for responding to optmization queries. It
-   * spawns a subsolver SmtEngine that captures the parent assertions and 
-   * implements a linear optimization loop. Supports activateObjective, 
-   * checkOpt, and objectiveGetValue in that order.
-   */
-  class OptimizationSolver
-  {
-   public:
-    /** parent is the smt_solver that the user added their assertions to **/
-    OptimizationSolver(SmtEngine* parent);
-    ~OptimizationSolver();
-
-    /** Runs the optimization loop for the activated objective **/
-    OptResult checkOpt();
-    /** Activates an objective: will be optimized for **/
-    void activateObj(const Node& obj, const int& type);
-    /** Gets the value of the optimized objective after checkopt is called **/
-    Node objectiveGetValue();
-
-   private:
-    /** The parent SMT engine **/
-    SmtEngine* d_parent;
-    /** The objectives to optimize for **/
-    Objective d_activatedObjective;
-    /** A saved value of the objective from the last sat call. **/
-    Node d_savedValue;
-  };
+
+  /** Specify whether to use signed or unsigned comparison
+   * for BitVectors (only for BitVectors), this variable is defaulted to true
+   * **/
+  bool d_signed;
+};
+
+
+/**
+ * A solver for optimization queries.
+ *
+ * This class is responsible for responding to optmization queries. It
+ * spawns a subsolver SmtEngine that captures the parent assertions and
+ * implements a linear optimization loop. Supports activateObjective,
+ * checkOpt, and objectiveGetValue in that order.
+ */
+class OptimizationSolver
+{
+ public:
+  /** parent is the smt_solver that the user added their assertions to **/
+  OptimizationSolver(SmtEngine* parent);
+  ~OptimizationSolver();
+
+  /** Runs the optimization loop for the activated objective **/
+  OptResult checkOpt();
+  /** Activates an objective: will be optimized for
+   * Parameter is_signed specifies whether we should use signed/unsigned
```

Comment:
1. the parameter name should match the real parameter name (is_signed vs. bv_is_signed_compare).
2. As above, the style of naming should be consistent with the rest of the repo.
3. When specifying parameters, this style should be used: https://github.com/CVC4/CVC4/blob/6beb70fcedd18e965ad82949090365cb44a43692/src/preprocessing/passes/bv_to_int.h#L117

Reply:
Thanks for addressing comment 2 here, but comments 1+3 still need to be addressed.

Reply:
this should have been resolved???

Reply:
yes -- 1 means that the name of the parameter in the comment should match the name of the parameter in the declaration, and 3 is just trying to be consistent with parts of the code that document parameters. Not all of the code is doing it that way, but I think it is a very clear way to document parameters and return values, and we are trying to use this style more.

Reply:
now it should have been resolved...

Reply:
resolved.